spec/mongoid/association/referenced/has_many/enumerable_spec.rb

```diff
@@ -201,6 +201,122 @@
     end
   end
 
+  describe "#empty?" do
+
+    let(:person) do
+      Person.create
+    end
+
+    let!(:post_one) do
+      Post.create(person_id: person.id)
+    end
+
+    let!(:post_two) do
+      Post.create(person_id: person.id)
+    end
+
+    context "when only a criteria target exists" do
+
+      let(:criteria) do
+        Post.where(person_id: person.id)
+      end
+
+      let!(:enumerable) do
+        described_class.new(criteria)
+      end
+
+      let!(:empty) do
+        enumerable.empty?
+      end
+
+      it "returns false" do
+        expect(empty).to be false
+      end
+
+      it "retains the correct length" do
+        expect(enumerable.length).to eq(2)
```

Comment:
If you intended to call `enumerable.empty?` and then perform additional assertions, these examples should be moved under their own context which has a `before` block which calls `.empty?`, and the context should be labeled something like "after .empty? is called".

Reply:
OK, I've changed it. Again I copied an existing spec.